tests/raftstore/test_conf_change.rs

```diff
@@ -425,3 +420,40 @@ fn test_server_after_remove_itself() {
     let mut cluster = new_server_cluster(0, count);
     test_after_remove_itself(&mut cluster);
 }
+
+fn test_add_remove_add<T: Simulator>(cluster: &mut Cluster<T>) {
+    init_log();
+    let r1 = cluster.bootstrap_conf_change();
+    cluster.start();
+    cluster.change_peer(r1, ConfChangeType::AddNode, 2);
+    cluster.must_put(b"k1", b"v1");
+    cluster.change_peer(r1, ConfChangeType::AddNode, 3);
+    must_get_equal(&cluster.get_engine(3), b"k1", b"v1");
+
+    let s1 = (1..3).collect();
+    let s2 = (3..4).collect();
+    cluster.partition(Arc::new(s1), Arc::new(s2));
+
+    // during partition, peer 3 don't know it's removed and add again
+    cluster.change_peer(r1, ConfChangeType::RemoveNode, 3);
+    cluster.change_peer(r1, ConfChangeType::AddNode, 3);
+
+    // when network recover, peer 3 will receive heartbeat
+    cluster.reset_transport_hooks();
+    print!("--------------------------------------------\n");
+    sleep_ms(3000);
```

Comment:
I don't think we need to sleep 3 secs here. Maybe we can `reset_leader` then `must_put` to verify raft has panic.